account_product_fiscal_classification/model/product_template.py

```diff
@@ -0,0 +1,105 @@
+
+    # Field Section
+    fiscal_classification_id = fields.Many2one(
+        comodel_name='account.product.fiscal.classification',
+        string='Fiscal Classification',
+        help="Specify the combination of taxes for this product."
+        " This field is required. If you dont find the correct Fiscal"
+        " Classification, Please create a new one or ask to your account"
+        " manager if you don't have the access right.")
+
+    # Overload Section
+    @api.model
+    def create(self, vals):
+        res = super(ProductTemplate, self).create(vals)
+        res.write_taxes_setting(vals)
+        return res
+
+    @api.multi
+    def write(self, vals):
+        res = super(ProductTemplate, self).write(vals)
+        self.write_taxes_setting(vals)
+        return res
+
+    # View Section
+    def fields_view_get(
+            self, cr, uid, view_id=None, view_type='form', context=None,
+            toolbar=False, submenu=False):
+        """Set 'fiscal_classification_id' as required by fields_view_get:
+        We don't set it by fields declaration in python file, to avoid
+        incompatibility with other modules that could have demo data
+        without fiscal_classification_id (the bugs will occur depending
+        of the order in which the modules are loaded);
+        We don't set it by view inheritance in xml file to impact all views
+        (form / tree) that could define the model 'product.template';"""
+        res = super(ProductTemplate, self).fields_view_get(
+            cr, uid, view_id=view_id, view_type=view_type, context=context,
+            toolbar=toolbar, submenu=submenu)
+        if view_type == 'form':
+            doc = etree.XML(res['arch'])
+            nodes = doc.xpath("//field[@name='fiscal_classification_id']")
+            if nodes:
+                nodes[0].set('required', '1')
+                setup_modifiers(
+                    nodes[0], res['fields']['fiscal_classification_id'])
+                res['arch'] = etree.tostring(doc)
+        return res
+
+    # Custom Section
+    def write_taxes_setting(self, vals):
+        """If Fiscal Classification is defined, set the according taxes
+        to the product(s); Otherwise, find the correct Fiscal classification,
+        depending of the taxes, or create a new one, if no one are found."""
+        if vals.get('fiscal_classification_id', False):
+            # update or replace 'taxes_id' and 'supplier_taxes_id'
+            classification = self.fiscal_classification_id
+            tax_vals = {
+                'supplier_taxes_id': [[6, 0, [
+                    x.id for x in classification.purchase_tax_ids]]],
+                'taxes_id': [[6, 0, [
+                    x.id for x in classification.sale_tax_ids]]],
+                }
+            super(ProductTemplate, self.sudo()).write(tax_vals)
+        elif 'supplier_taxes_id' in vals.keys() or 'taxes_id' in vals.keys():
+            # product template Single update mode
```

Comment:
* all cases : While importing data. (see description of the module);
https://github.com/OCA/account-fiscal-rule/pull/28/files#diff-39aa8614d5d8e95ba38834e7370a8b07R34
* custom cases : if a custom module set taxes not readonly; (to all user or some specific like accountant users)